lib/puppet/provider/ios_ntp_access_group/cisco_ios.rb

```diff
@@ -0,0 +1,78 @@
+require 'puppet/resource_api/simple_provider'
+require_relative '../../util/network_device/cisco_ios/device'
+require_relative '../../../puppet_x/puppetlabs/cisco_ios/utility'
+
+# NTP Access Group Puppet Provider for Cisco IOS devices
+class Puppet::Provider::IosNtpAccessGroup::CiscoIos < Puppet::ResourceApi::SimpleProvider
+  def self.commands_hash
+    @commands_hash ||= PuppetX::CiscoIOS::Utility.load_yaml(File.expand_path(__dir__) + '/command.yaml')
+  end
+
+  def self.instances_from_cli(output)
+    new_instance_fields = []
+    output.scan(%r{#{PuppetX::CiscoIOS::Utility.get_instances(commands_hash)}}).each do |raw_instance_fields|
+      new_instance = PuppetX::CiscoIOS::Utility.parse_resource(raw_instance_fields, commands_hash)
+      new_instance[:ipv6_access_group] = (new_instance[:ipv6_access_group]) ? true : false
+      new_instance[:ensure] = 'present'
+      new_instance_fields << new_instance
+    end
+    new_instance_fields
+  end
+
+  def self.commands_from_instance(instance)
+    commands = []
+    if instance[:name].casecmp('none').zero?
+      instance[:name] = nil
+    end
+    instance[:access_group_type] = 'ipv6 ' + instance[:access_group_type] if instance[:ipv6_access_group]
+    command = PuppetX::CiscoIOS::Utility.set_values(instance, commands_hash)
+    if instance[:ensure].to_s == 'absent'
+      command = 'no ' + command
+    end
+    commands << command
+    puts commands
+    commands
+  end
+
+  def commands_hash
+    Puppet::Provider::IosNtpAccessGroup::CiscoIos.commands_hash
+  end
+
+  def get(context, _names = nil)
+    output = context.transport.run_command_enable_mode(PuppetX::CiscoIOS::Utility.get_values(commands_hash))
+    return [] if output.nil?
+    return_value = Puppet::Provider::IosNtpAccessGroup::CiscoIos.instances_from_cli(output)
+    PuppetX::CiscoIOS::Utility.enforce_simple_types(context, return_value)
+  end
+
+  def create(context, name, should)
+    create_hash = { name: name, ensure: 'create', access_group_type: should[:access_group_type], ipv6_access_group: should[:ipv6_access_group] }
```

Comment:
What is `create` in the `ensure`? Is there a particular reason - can be confusing as `ensure` is `present` or `absent`?